libs/providers/go-feature-flag-web/src/lib/go-feature-flag-web-provider.ts

```diff
@@ -0,0 +1,341 @@
+    while (attempt < this._maxRetries) {
+      attempt++;
+      await this.connectWebsocket()
+      if (this._websocket !== undefined && this._websocket.readyState === WebSocket.OPEN) {
+        return
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, delay));
+      delay *= this._retryDelayMultiplier;
+      this._logger?.info(`error while reconnecting the websocket, next try in ${delay} ms (${attempt}/${this._maxRetries}).`)
+    }
+    this.events.emit(ProviderEvents.Stale, {
+      message: 'impossible to get status from GO Feature Flag (websocket connection stopped)'
+    });
+  }
+
+  onClose(): Promise<void> {
+    this._websocket?.close(1000, "Closing GO Feature Flag provider");
+    return Promise.resolve();
+  }
+
+  async onContextChange(_: EvaluationContext, newContext: EvaluationContext): Promise<void> {
+    this._logger?.debug(`new context provided: ${newContext}`);
+    this.events.emit(ProviderEvents.Stale, {message: 'context has changed'});
+    await this.retryFetchAll(newContext);
+    this.events.emit(ProviderEvents.Ready, {message: ''});
+  }
+
+  resolveNumberEvaluation(flagKey: string): ResolutionDetails<number> {
+    return this.evaluate(flagKey, 'number')
+  }
+
+  resolveObjectEvaluation<T extends FlagValue>(flagKey: string): ResolutionDetails<T> {
+    return this.evaluate(flagKey, 'object')
+  }
+
+  resolveStringEvaluation(flagKey: string): ResolutionDetails<string> {
+    return this.evaluate(flagKey, 'string')
+  }
+
+  resolveBooleanEvaluation(flagKey: string): ResolutionDetails<boolean> {
+    return this.evaluate(flagKey, 'boolean')
+  }
+
+  private evaluate<T extends FlagValue>(flagKey: string, type: string): ResolutionDetails<T> {
+    const resolved = this._flags[flagKey];
+    if (!resolved) {
+      throw new FlagNotFoundError(`flag key ${flagKey} not found in cache`);
+    }
+
+    if (typeof resolved.value !== type) {
+      throw new TypeMismatchError(`flag key ${flagKey} is not of type ${type}`);
+    }
+    return {
+      variant: resolved.variant,
+      value: resolved.value as T,
+      flagMetadata: resolved.flagMetadata,
+      errorCode: resolved.errorCode,
+      errorMessage: resolved.errorMessage,
+      reason: this._websocket?.readyState !== WebSocket.OPEN ? StandardResolutionReasons.CACHED : resolved.reason,
+    };
+  }
+
+  private async retryFetchAll(ctx: EvaluationContext, flagsChanged: string[] = []) {
+    let delay = this._retryInitialDelay;
+    let attempt = 0;
+    while (attempt < this._maxRetries) {
+      attempt++;
+      try {
+        await this.fetchAll(ctx, flagsChanged);
+        this._status = ProviderStatus.READY;
+        return
+      } catch (err) {
+        this._status = ProviderStatus.ERROR;
+        this.handleFetchErrors(err)
+        await new Promise((resolve) => setTimeout(resolve, delay));
+        delay *= this._retryDelayMultiplier;
+        this._logger?.info(`xxx ${delay} ms (${attempt}/${this._maxRetries}).`)
```

Comment:
Is this "xxx" supposed to be here?

Something I like to do is use the classname as a prefix to my log messages: `${GoFeatureFlagWebProvider.name}: ....`